classes/quiz_settings.php

```diff
@@ -402,4 +451,67 @@ private function split_keys(string $keys) : array {
         }
         return $keys;
     }
+
+    /**
+     * Try and get a file in the user draft filearea by itemid.
+     *
+     * @param string $itemid Item ID of the file.
+     * @return stored_file|null Returns null if no file is found.
+     *
+     * @throws \coding_exception
+     */
+    private function get_current_user_draft_file(string $itemid) : ?stored_file { // @codingStandardsIgnoreLine
+        global $USER;
+        $context = context_user::instance($USER->id);
+        $fs = get_file_storage();
+        if (!$files = $fs->get_area_files($context->id, 'user', 'draft', $itemid, 'id DESC', false)) {
+            return null;
+        }
+        return reset($files);
+    }
+
+    /**
+     * Saves filemanager_sebconfigfile files to the moodle storage backend.
+     *
+     * @return bool
+     * @throws \coding_exception
+     */
+    private function save_filemanager_sebconfigfile_draftarea() : bool {
+        $draftitemid = file_get_submitted_draft_itemid('filemanager_sebconfigfile');
```

Comment:
let's refactor this and move file uploading to the form processing part of the code. This will decouple persistent from the form submitting. I recon we can just pass file/fileid to persistent